src/mds/MDCache.cc

```diff
@@ -13256,5 +13302,30 @@ void MDCache::handle_mdsmap(const MDSMap &mdsmap) {
     it = q.erase(it);
     in->maybe_export_pin();
   }
+
+  /* Handle consistent hash ring during cluster resizes */
+  if (mdsmap.get_max_mds() != oldmap.get_max_mds()) {
+        dout(10) << "Max mds is increased. Noted from MDCache side" << dendl;
+        elist<CInode*>::iterator it = ephemeral_pins.begin(member_offset(CInode, ephemeral_pin_inode));
+        while (!it.end()) {
+          // Migrate if the inodes hash elsewhere
+          if (hash_into_rank_bucket((*it)->ino(), mdsmap.get_max_mds()) != mds->get_nodeid()) {
```

Comment:
The duplication of logic here with `maybe_export_ephemeral_*` should be avoidable. Let's make a method:

`CInode::queue_ephemeral_export()`

That takes an inode that **is** ephemerally pinned already and tries to export it to the right place. If it's already in the right place, it removes it from the queue.